app/javascript/events.js

```diff
@@ -1,4 +1,4 @@
-let cachedEvents = JSON.parse(localStorage.getItem("cachedEvents")) || [];
```

Comment:
Without the let keyword you are not creating and initializing a new variable. Does this work?

Reply:
this fixed the error "cachedEvents has already been define".

Reply:
Where is this error coming from? Also, does the code work when first initializing?

Reply:
if you inspect the webpage, and go to console. you'll see those. It works on my side. you can also test this in user_login branch.